### PR TITLE
Add tests for nested props and refs for field helper

### PR DESCRIPTION
**Pull request by zemccartney**
Added a test for refs, as we discussed. Also randomly felt compelled to throw in a test on nested properties; didn't know if it was desirable, just felt curious. 

🙏 🍺

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -1601,6 +1601,68 @@ describe('Schwifty', () => {
                 expect(c.validate('x')).to.equal({ value: 'x' });
                 expect(c.validate(1)).to.contain('error');
             });
+
+            it('supports nested properties', () => {
```
Comment (devinivy) [inline test/index.js]:
```suggestion
            it('supports nested properties.', () => {
```

File: test/index.js [hunk 2]
```diff
@@ -1601,6 +1601,68 @@ describe('Schwifty', () => {
...
+
+                expect(dfull.validate('123')).to.equal({ value: '123' });
+                expect(dfull.validate('12')).to.contain('error');
+
+                expect(efull.validate()).to.equal({ value: 'e' });
+                expect(efull.validate('x')).to.equal({ value: 'x' });
+                expect(efull.validate(1)).to.contain('error');
+            });
+
+            it('throws when called on a field whose schema contains a ref', () => {
```
Comment (devinivy) [inline test/index.js]: I wonder if it would be nice to rephrase this to indicate that the call to `field()` doesn't actually throw.  Validation also doesn't _necessarily_ throw, e.g. if the field is plopped into a larger schema where the ref does exist.  This schema could work just fine, for example:

```js
Joi.object({
    a: Model.field('a'),
    c: Model.field('c')
});
```

**Comment by devinivy** [comment]
Thanks for this!

File: test/index.js (hunk 2 above)
Comment (zemccartney) [reply test/index.js]: Woah, no way! I didn't realize that was possible, thanks for pointing that technique out.
And for the suggested rewording, that makes sense. I just pushed up some changes that hopefully addressed those points 🙏

File: test/index.js [hunk 3]
```diff
@@ -1601,6 +1601,74 @@ describe('Schwifty', () => {
...
+
+                expect(dfull.validate('123')).to.equal({ value: '123' });
+                expect(dfull.validate('12')).to.contain('error');
+
+                expect(efull.validate()).to.equal({ value: 'e' });
+                expect(efull.validate('x')).to.equal({ value: 'x' });
+                expect(efull.validate(1)).to.contain('error');
+            });
+
+            it('validation throws when the schema contains an invalid ref', () => {
```
Comment (devinivy) [inline test/index.js]:
```suggestion
            it('validation throws when the schema contains an invalid ref.', () => {
```
